lib/rialto/etl/cli/composite_etl.rb

```diff
@@ -106,18 +107,27 @@ def file_prefix
           raise NotImplementedError
         end
 
+        # rubocop:disable Metrics/MethodLength
         def extract(row, profile_id, force)
           extract_file = File.join(input_directory, "#{file_prefix}-#{profile_id}.ndj")
           if File.exist?(extract_file) && !force
             say "file #{extract_file} already exists, skipping. use -f to force overwrite"
             return extract_file
           end
-          results = perform_extract(row)
+          # Don't write file if an exception occurs
+          begin
+            results = perform_extract(row)
+          rescue StandardError
+            say "Skipping #{extract_file} because an error occurred while extracting"
+            return extract_file
```

Comment:
We're returning a name to a non-existent file?

Reply:
Yes, the code that invokes this checks to see if the file exists and is not empty.

An empty file is a normal result of this method, as it indicates that there are no publications.